eslint.config.mjs

```diff
@@ -0,0 +1,403 @@
+
+          groups: [
+            'builtin',
+            'external',
+            'internal',
+            'parent',
+            ['index', 'sibling'],
+            'object',
+          ],
+
+          pathGroups: [
+            {
+              pattern: '{react,react-dom,react-dom/client,prop-types}',
+              group: 'builtin',
+              position: 'after',
+            },
+            {
+              pattern: '{react-intl,intl-messageformat}',
+              group: 'builtin',
+              position: 'after',
+            },
+            {
+              pattern:
+                '{classnames,react-helmet,react-router,react-router-dom}',
+              group: 'external',
+              position: 'before',
+            },
+            {
+              pattern:
+                '{immutable,@reduxjs/toolkit,react-redux,react-immutable-proptypes,react-immutable-pure-component}',
+              group: 'external',
+              position: 'before',
+            },
+            {
+              pattern: '{mastodon/**}',
+              group: 'internal',
+              position: 'after',
+            },
+          ],
+
+          pathGroupsExcludedImportTypes: [],
+        },
+      ],
+
+      'promise/always-return': 'off',
+      'promise/catch-or-return': [
+        'error',
+        {
+          allowFinally: true,
+        },
+      ],
+      'promise/no-callback-in-promise': 'off',
+      'promise/no-nesting': 'off',
+      'promise/no-promise-in-callback': 'off',
+
+      'react/jsx-filename-extension': [
+        'error',
+        {
+          extensions: ['.jsx', 'tsx'],
+        },
+      ],
+
+      'react/jsx-boolean-value': 'error',
+      'react/display-name': 'off',
+      'react/jsx-fragments': ['error', 'syntax'],
+      'react/jsx-equals-spacing': 'error',
+      'react/jsx-no-bind': 'error',
+      'react/jsx-no-useless-fragment': 'error',
+      'react/jsx-no-target-blank': [
+        'error',
+        {
+          allowReferrer: true,
+        },
+      ],
+      'react/jsx-tag-spacing': 'error',
+      'react/jsx-wrap-multilines': 'error',
+      'react/self-closing-comp': 'error',
+    },
+  },
+  {
+    files: [
+      'app/javascript/mastodon/common.js',
+      'app/javascript/mastodon/features/emoji/unicode_to_unified_name.js',
+      'app/javascript/mastodon/features/emoji/emoji_compressed.js',
+      'app/javascript/mastodon/features/emoji/unicode_to_filename.js',
+      'app/javascript/mastodon/service_worker/web_push_locales.js',
+      '**/*.config.js',
+      '**/.*rc.js',
+      '**/ide-helper.js',
+      'config/webpack/**/*',
+      'config/formatjs-formatter.js',
+    ],
+
+    languageOptions: {
+      globals: {
+        ...globals.commonjs,
+        ...globals.node,
+      },
+
+      ecmaVersion: 5,
+      sourceType: 'commonjs',
+    },
+
+    rules: {
+      'import/no-commonjs': 'off',
+    },
+  },
+  {
+    files: ['**/*.ts', '**/*.tsx'],
+
+    extends: [
+      js.configs.recommended,
```

Comment:
Do we really need to list all plugins here? If they are in the extended config list, then we should not have to add them to `extends`?